radix-engine/src/system/transaction/instructions.rs

```diff
@@ -318,6 +319,38 @@ impl TxnNormalInstruction for AssertWorktopContainsNonFungibles {
     }
 }
 
+impl TxnNormalInstruction for AssertNextCallReturnsOnly {
+    fn execute<Y: SystemApi<RuntimeError> + KernelNodeApi + KernelSubstateApi<L>, L: Default>(
+        self,
+        _worktop: &mut Worktop,
+        objects: &mut IntentProcessorObjects,
+        _api: &mut Y,
+    ) -> Result<InstructionOutput, RuntimeError> {
+        objects.next_call_return_constraints = Some(NextCallReturnConstraints {
```

Comment:
For consistency, perhaps this should be `next_call_returns_constraints = Some(NextCallReturnsConstraints` ?

Reply:
Perhaps we should panic or throw an error if this is already set when we set it?

It shouldn't be possible (the validation should enforce it) but sometimes we run transactions without validation (e.g. system transactions).

Reply:
I think it's fine behavior that the system will just override the last setting.